### PR TITLE
feat(app): module slideout QA feedback touchups 

### LINKED ISSUE #9869: Address Slideout Design QA Feedback
# Overview

This ticket captures design QA feedback for slideouts in App

# Implementations Details
- Review and update designs for slideout per the purple text in this figma. There are before and after screenshots. Basically, the end result should look like the "after" screenshot. https://www.figma.com/file/bVtCogdwiXj3ndpk4AWXCx/Milestone-1%3A-Read-and-Execute?node-id=8491%3A271066

**General** 
- Update text color to proper specified color in design guidelines
- Update all button text (main CTA) to "Confirm" for module settings
- Add "Close" primary CTA button for all "About" module slideouts
- All input components: remove the < > buttons 
- Add hover and pressed state for upper right "X" (cancel) button
- Ensure that acceptable range is always displayed under the input field -- "input out of range" information should not replace that, but appear below the range
- Heater-Shaker should be referenced as "Heater-Shaker" -- find and replace other instances (e.g., "Heater Shaker," "Heater/Shaker") 
- If target is N/A, there should be no unit displayed next to N/A (e.g., remove "RPM" if N/A)

**Specific Slideouts**
- H-S Set temp: Update space to be 16 px between button and title
- H-S Set temp: Title should grow heightwise
- H-S Set temp: Update temp range to 37 - 95 C. 37 C is lowest
- H-S Test Shake: Update margin left in top gray box
- H-S Test Shake: Update to new info icon
- H-S Test Shake: Ensure font style is p semibold in upper gray box
- H-S Test Shake: Center align labware latch 
- H-S Test Shake: Diff disabled shake speed
- H-S Test Shake: Make "Troubleshooting" text dark black and update and center dropdown icon
- Mag engage: Update "Gen 1 Height Ranges" and "Engage Height" title color
- Mag engage: Make sure .5rem spacing between list items
- Mag engage: Add unit for engage height input field
- TC Set block temp: Update input field title text to say "Set block temperature"
- TC Set lid temp: Update input field title text to say "Set lid temperature"

# Design Links
- https://www.figma.com/file/bVtCogdwiXj3ndpk4AWXCx/Milestone-1%3A-Read-and-Execute?node-id=8491%3A271066

# Acceptance Criteria
- [x]  Review and update designs for slideout per the purple text in this figma. There are before and after screenshots. Basically, the end result should look like the "after" screenshot. 

**General** 
- [x] Update text color to proper specified color in design guidelines
- [x] Update all button text (main CTA) to "Confirm" for module settings
- [x] Add "Close" primary CTA button for all "About" module slideouts
- [x] All input components: remove the < > buttons 
- [x] Add hover and pressed state for upper right "X" (cancel) button
- [x] Ensure that acceptable range is always displayed under the input field -- "input out of range" information should not replace that, but appear below the range
- [x] Heater-Shaker should be referenced as "Heater-Shaker" -- find and replace other instances (e.g., "Heater Shaker," "Heater/Shaker") 
- [x] If target is N/A, there should be no unit displayed next to N/A (e.g., remove "RPM" if N/A)

**Specific Slideouts**
- [x] H-S Set temp: Update space to be 16 px between button and title
- [ ] H-S Set temp: Title should grow heightwise
- [x] H-S Set temp: Update temp range to 37 - 95 C. 37 C is lowest
- [x] H-S Test Shake: Update margin left in top gray box
- [x] H-S Test Shake: Update to new info icon
- [x] H-S Test Shake: Ensure font style is p semibold in upper gray box
- [x] H-S Test Shake: Center align labware latch 
- [x] H-S Test Shake: Diff disabled shake speed
- [x] H-S Test Shake: Make "Troubleshooting" text dark black and update and center dropdown icon
- [x] Mag engage: Update "Gen 1 Height Ranges" and "Engage Height" title color
- [x] Mag engage: Make sure .5rem spacing between list items
- [x] Mag engage: Add unit for engage height input field
- [x] TC Set block temp: Update input field title text to say "Set block temperature"
- [x] TC Set lid temp: Update input field title text to say "Set lid temperature"

### PR DESCRIPTION
**Pull request by jerader**
closes #9869 

# Overview

This addresses the module slideout QA feedback including a few misc items such as changing Heater-Shaker displayname to include the hypen.

# Changelog

- updated module slideouts including about module slideouts and the test shake slideout
- created new `Collapsible` component in the `moduleCard` folder so it can be shared between both the `TestShake` component and the `TestShakeSlideout` component
- added a few states to the cancel icon in the `slideout` component
- updated the `information` icon to one matching the UI overhaul
- changed Heater-Shaker display name

# Review requests

- do the slideouts match designs?

# Risk assessment

low, behind ff

### CODE REVIEW

**Comment by mmencarelli** [comment]
**Round 1 Feedback on all module/pipette slideouts from the T2 Devices page**

--

Link to the [Figma frame](https://www.figma.com/file/bVtCogdwiXj3ndpk4AWXCx/Milestone-1%3A-Read-and-Execute?node-id=11435%3A270267) with annotations and side by side comparison of code + design

**About module slideouts**
- [x] (1) Update font colour to our 'dark-black' colour
- [x] (2) Need the other states for the 'x'
- [x] (3) Update the colour to our 'dark-grey/enabled', make sure that 1% letter-spacing is in there too
- [x] (4) Update the p colour to 'dark-black'
- [x] (5) Add the 'close' button into all 'about' slideouts. This was a recent addition! 
- [ ] (6) If it takes less than 10 mins, could you add the reverse transition you have on the slideout for all slideouts? I love how it slides out from the right and would love to see it slide back to the right upon closing! Not a biggie though

**Module setting slideouts**
- [x] (1) Update font colour to our 'dark-black' colour
- [x] (2) Need the other states for the 'x'
- [x] (3) Update the p colour to 'dark-black'
- [x] (4) This label should be updated to our 'label-semibold' styling. So upping the font size, and also making sure that colour is our 'dark-black'. There's also a margin-bottom of 8px on the label. Currently there's 2px!
> Matching the input styles with our new component! Breakdown on loom. 
- [ ] (5) Update to our label styling (font size goes bigger)
- [x] (6) label disappears on error state
- [x] (7) Missing focus state? 
- [x] (8) Update the border colour on the selected/active state
- [ ] (9) Update button styling to the primary button style (including all the states!)
- [x] (10) Update colour to match design
- [x] (11) There's an extra 8px padding below the final list item
- [x] (12) Add a margin-right of 8px to the title so copy doesn't get too close to the 'x'

**Pipette Settings**
- [ ] (1) Update font colour to our 'dark-black' colour
- [ ] (2) Need the other states for the 'x'
- [ ] (3) Add a margin-right: 8px to the title of the slideout **this is across all slideouts actually**
- [ ] (4) Update to our p class styling
- [ ] (5) Update to our secondary button styling (including all states, etc.)
- [ ] (6) Can remove the : after the headers in these spots
- [ ] (7) Should be a margin-bottom of 8px on the _form_group_label, also make sure this is the dark-black colour!
- [ ] (8) Label needs to be updated in colour, and semibold
- [ ] (9) Just make sure the input fields are up to date with the current design. Currently need some updating! 
- [ ] (10) Add in that styling when content goes beyond the height of the viewport. CTA should have that white background with a light drop shadow behind it. 

**Test Shake**
- [x] (1) Update font colour to our 'dark-black' colour
- [x] (2) Need the other states for the 'x'
- [x] (3) Not sure if this will be handled in other work, but the 'labware latch' column should actually be stacked beneath 'shaker' column! 
- [x] (4) Can't test this myself, but it looks like both of these labels are a bit bigger than our 'label' style 
- [x] (5) Update the disabled state of the tertiary button to match design
- [x] (6) When content overflows the height of the viewport, use the white background + dropshadow version of the CTA. Displayed to the right! 


Screen Shot 2022-05-05 at 10 48 59 AM
Screen Shot 2022-05-05 at 10 49 04 AM
Screen Shot 2022-05-05 at 10 49 22 AM
Screen Shot 2022-05-05 at 10 49 30 AM
Screen Shot 2022-05-05 at 10 49 39 AM

**Comment by jerader** [comment]
The above comment with design feedback, the Pipette Settings slideout section will be completed in a follow up ticket since this PR is focusing on module slideouts.

The follow up ticket is: https://github.com/Opentrons/opentrons/issues/10228

File: app/src/organisms/Devices/ModuleCard/Collapsible.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        alignItems={ALIGN_CENTER}
+        onClick={toggleExpanded}
+      >
+        <Flex flexDirection={DIRECTION_COLUMN}>
+          <StyledText as="h5">{title}</StyledText>
+        </Flex>
+        <Icon
+          size={SIZE_1}
+          name={expanded ? collapse : expand}
+          aria-label={`expandedIcon` ?? 'plus'}
```
Comment (shlokamin) [inline app/src/organisms/Devices/ModuleCard/Collapsible.tsx:61]: this static nullish coalesce will always result in `expandedIcon `

**Review by shlokamin — APPROVED** [review]
code changes LGTM!

**Pull request merged**